precompile/contracts/rewardmanager/contract_test.go

```diff
@@ -341,3 +345,116 @@ func TestRewardManagerRun(t *testing.T) {
 func BenchmarkRewardManager(b *testing.B) {
 	allowlist.BenchPrecompileWithAllowList(b, Module, state.NewTestStateDB, tests)
 }
+
+func TestSetRewardAddressLogging(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+
+	blockContext := contract.NewMockBlockContext(ctrl)
+	blockContext.EXPECT().Number().Return(big.NewInt(0)).AnyTimes()
+	blockContext.EXPECT().Timestamp().Return(uint64(time.Now().Unix())).AnyTimes()
+
+	baseState := state.NewTestStateDB(t)
+	accessibleState := contract.NewMockAccessibleState(ctrl)
+	accessibleState.EXPECT().GetStateDB().Return(baseState).AnyTimes()
+	accessibleState.EXPECT().GetBlockContext().Return(blockContext).AnyTimes()
+
+	allowlist.SetAllowListRole(baseState, Module.Address, allowlist.TestAdminAddr, allowlist.AdminRole)
+
+	rewardAddress = common.HexToAddress("0x0123")
+	input, err := PackCurrentRewardAddressOutput(rewardAddress)
+	require.NoError(err)
+
+	_, _, err = setRewardAddress(
+		accessibleState,
+		allowlist.TestAdminAddr,
+		Module.Address,
+		input,
+		SetRewardAddressGasCost,
+		false,
+	)
+	require.NoError(err)
+
+	// Check logs are stored in state
+	expectedTopic := []common.Hash{
+		RewardManagerABI.Events["RewardAddress"].ID,
+		allowlist.TestAdminAddr.Hash(),
+		rewardAddress.Hash(),
+	}
+
+	allLogs := baseState.(*state.StateDB).Logs()
+	require.Len(allLogs, 1)
+	require.Equal(expectedTopic, allLogs[0].Topics)
+}
```

Comment:
Check sad path too? if setRewardAddress fails, we should not expect event
?

Reply:
@anusha-ctrl not sure what the sad path is here? Log serialization can really fail only if there is bug. I introduced a test to show no logs are stored pre-fork. Not sure what else should I cover, except for paying pre-fork gas cost for ops involving logs?